disnake/ext/commands/interaction_bot_base.py

```diff
@@ -724,42 +964,39 @@ def decorator(
                 extras=extras,
                 **kwargs,
             )(func)
-            self.add_message_command(result)
+            self.add_app_command(result)
             return result
 
         return decorator
 
     # command synchronisation
 
     def _ordered_unsynced_commands(
-        self, test_guilds: Optional[Sequence[int]] = None
+        self,
     ) -> Tuple[List[ApplicationCommand], Dict[int, List[ApplicationCommand]]]:
-        global_cmds = []
-        guilds = {}
+        global_cmds: List[ApplicationCommand] = []
+        guilds: Dict[int, List[ApplicationCommand]] = {}
 
-        for cmd in self.application_commands_iterator():
+        for key, cmd in self.all_app_commands.items():
             if not cmd.auto_sync:
                 cmd.body._always_synced = True
 
-            guild_ids = cmd.guild_ids or test_guilds
```

Comment:
mentioned this [here](https://canary.discord.com/channels/808030843078836254/913779868985090089/1154798090071592970) a while ago, just found it again -

In the previous implementation, `@slash_command(guild_ids=[])` with `test_guilds` set would result in the command being registered in those `test_guilds`, since the code here used `cmd.guild_ids or test_guilds`.
The new logic (see previous comment) now checks `cmd.guild_ids is None`, which means such commands won't be registered at all now.

I'm a bit undecided on which behavior makes more sense (probably the former? not seeing a use case for `guild_ids=[]`, I'd say it should be equivalent to `guild_ids=None`), we may want to consider reverting this to the old logic at least until v3

Reply:
Good catch

Reply:
Fixed this in https://github.com/DisnakeDev/disnake/pull/1107/commits/55299b25c5a52bddbd2d2987943aa590b6400dc6